apps/ledger-live-desktop/src/renderer/screens/customImage/Step4Transfer.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useEffect, useState } from "react";
+import { Alert, Flex, Text } from "@ledgerhq/react-ui";
+import { ProcessorResult } from "~/renderer/components/CustomImage/ImageGrayscalePreview";
+import { Step, StepProps } from "./types";
+import { useTranslation } from "react-i18next";
+import StepFooter from "./StepFooter";
+import { targetDimensions } from "~/renderer/components/CustomImage/shared";
+import { ImageBase64Data } from "~/renderer/components/CustomImage/types";
+import { createCanvas } from "~/renderer/components/CustomImage/imageUtils";
+import { ImageProcessingError } from "@ledgerhq/live-common/customImage/errors";
+import StepContainer from "./StepContainer";
+
+type Props = StepProps & {
+  result?: ProcessorResult;
+};
+
+function reconstructImage({
+  hexData,
+  width,
+  height,
+}: ProcessorResult["rawResult"]): ImageBase64Data {
+  const { canvas, context } = createCanvas();
+  canvas.height = height;
+  canvas.width = width;
+  if (context === null) throw new Error("Context is null");
+
+  const numLevelsOfGray = 16;
+  const rgbStep = 255 / (numLevelsOfGray - 1);
+
+  const pixels = Array.from(Array(height), () => Array(width));
+  hexData.split("").forEach((char, index) => {
```

Comment:
Maybe add a comment saying again that it's "by column, starting on the top right"